src/nodes/chunk_dispatch.c

```diff
@@ -145,6 +145,15 @@ ts_chunk_dispatch_get_chunk_insert_state(ChunkDispatch *dispatch, Point *point,
 		 * where the chunk already exists.
 		 */
 		Chunk *new_chunk = ts_hypertable_find_chunk_for_point(dispatch->hypertable, point);
+
+#if PG14_GE
+		/*
+		 * Frozen chunks require at least PG14.
+		 */
+		if (new_chunk && ts_chunk_is_frozen(new_chunk))
+			elog(ERROR, "cannot INSERT into frozen chunk");
```

Comment:
I think it would be good to print the chunk relname here also

Reply:
done